src/Sulu/Bundle/ContactBundle/Resources/public/js/components/contact-selection/main.js

```diff
@@ -0,0 +1,254 @@
+                                name: 'firstName'
+                            },
+                            {
+                                content: 'lastName',
+                                name: 'lastName'
+                            }
+                        ]
+                    }
+                }
+            ]);
+        },
+
+        /**
+         * Updates the datagrid when opening the overlay again
+         */
+        updateSnippetList = function() {
+            var selectedItems = this.getData() || [];
+
+            this.sandbox.emit(
+                'husky.datagrid.selected.update',
+                selectedItems
+            );
+        },
+
+        /**
+         * handle dom events
+         */
+        bindDomEvents = function() {
+            this.sandbox.dom.on(this.$el, 'click', function() {
+                return false;
+            }.bind(this), '.search-icon');
+
+            this.sandbox.dom.on(this.$el, 'keydown', function(e) {
+                if (event.keyCode === 13) {
+                    e.preventDefault();
+                    e.stopPropagation();
+
+                    return false;
+                }
+            }.bind(this), '.search-input');
+        },
+
+        /**
+         * starts the overlay component
+         */
+        startAddOverlay = function() {
+            var $element = this.sandbox.dom.createElement('<div/>');
+
+            this.sandbox.dom.append(this.$el, $element);
+            this.sandbox.start([
+                {
+                    name: 'overlay@husky',
+                    options: {
+                        triggerEl: this.$addButton,
+                        cssClass: 'snippet-content-overlay',
+                        el: $element,
+                        removeOnClose: false,
+                        container: this.$el,
+                        instanceName: 'contact-selection.' + this.options.instanceName + '.add',
+                        skin: 'wide',
+                        slides: [
+                            {
+                                title: this.sandbox.translate(this.options.translations.addContact),
+                                cssClass: 'snippet-content-overlay-add',
+                                data: templates.data(this.options),
+                                okCallback: getAddOverlayData.bind(this)
+                            }
+                        ]
+                    }
+                }
+            ]);
+        },
+
+        getAddOverlayData = function() {
+            this.sandbox.emit('husky.datagrid.items.get-selected', function(selected) {
+                this.setData(selected);
+            }.bind(this));
+        };
+
+    return {
+        type: 'itembox',
+
+        initialize: function() {
+            // extend default options
+            this.options = this.sandbox.util.extend(true, {}, defaults, this.options);
+
+            // init ids
+            this.options.ids = {
```

Comment:
I know we are doing this more often, but I think this is not a good idea... The options array should not be changed IMO, since passed options might be overridden.